.github/workflows/main.yml

```diff
@@ -0,0 +1,35 @@
+# This is a basic workflow to help you get started with Actions
+
+name: k8s-set-context
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ cluster-connect ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - uses: Azure/k8s-set-context@cluster-connect
+        with:
+          method: spn
+          cluster-type: 'arc'
+          creds: '${{ secrets.AZURE_CREDS }}'
+          cluster-name: arcaction
+          resource-group: atharvatest2
+          token: '${{ secrets.SA_TOKEN }}'
+      - run: |
```

Comment:
what is the ending run for ?.

Reply:
to run kubectl commands, so that we can validate the e2e flow.